staging/src/k8s.io/apiserver/pkg/features/kube_features.go

```diff
@@ -63,19 +65,47 @@ const (
 	// Allow API clients to retrieve resource lists in chunks rather than
 	// all at once.
 	APIListChunking utilfeature.Feature = "APIListChunking"
+
+	// owner: @apelisse
+	// alpha: v1.12
+	//
+	// Allow requests to be processed but not stored, so that
+	// validation, merging, mutation can be tested without
+	// committing.
+	DryRun utilfeature.Feature = "DryRun"
 )
 
 func init() {
-	utilfeature.DefaultFeatureGate.Add(defaultKubernetesFeatureGates)
+	// Only initialize global feature in the DefaultFeatureGate.
+	specs := map[utilfeature.Feature]utilfeature.FeatureSpec{}
+	for _, feature := range GlobalFeatures {
+		spec, ok := DefaultKubernetesFeatureGates[feature]
+		if !ok {
+			panic(fmt.Errorf("GlobalFeature %q not defined in DefaultKubernetesFeatureGates", feature))
+		}
+		specs[feature] = spec
+	}
+	utilfeature.DefaultFeatureGate.Add(specs)
 }
 
-// defaultKubernetesFeatureGates consists of all known Kubernetes-specific feature keys.
+// DefaultKubernetesFeatureGates consists of all known Kubernetes-specific feature keys.
 // To add a new feature, define a key for it above and add it here. The features will be
 // available throughout Kubernetes binaries.
-var defaultKubernetesFeatureGates = map[utilfeature.Feature]utilfeature.FeatureSpec{
+var DefaultKubernetesFeatureGates = map[utilfeature.Feature]utilfeature.FeatureSpec{
 	StreamingProxyRedirects: {Default: true, PreRelease: utilfeature.Beta},
 	AdvancedAuditing:        {Default: true, PreRelease: utilfeature.Beta},
 	APIResponseCompression:  {Default: false, PreRelease: utilfeature.Alpha},
 	Initializers:            {Default: false, PreRelease: utilfeature.Alpha},
 	APIListChunking:         {Default: true, PreRelease: utilfeature.Beta},
+	DryRun:                  {Default: false, PreRelease: utilfeature.Alpha},
+}
+
+// GlobalFeatures keeps track of features that can be accessed through
+// the DefaultFeatureGate global variable.
+var GlobalFeatures = []utilfeature.Feature{
```

Comment:
I am against this change in this PR, especially mixed in a commit to add another gate. This is a fundamental change of the feature gate logic. Can